js/packages/wallet-adapter-mobile/src/adapter.ts

```diff
@@ -234,16 +235,51 @@ export class SolanaMobileWalletAdapter extends BaseMessageSignerWalletAdapter {
     async sendTransaction(
         transaction: Transaction,
         connection: Connection,
-        _options?: SendOptions,
+        options?: SendOptions,
     ): Promise<TransactionSignature> {
         return await this.runWithGuard(async () => {
             const { authToken } = this.assertIsAuthorized();
             try {
                 return await this.transact(async (wallet) => {
-                    await this.performReauthorization(wallet, authToken);
+                    let targetCommitment: Finality;
+                    switch (connection.commitment) {
+                        case 'confirmed':
+                        case 'finalized':
+                        case 'processed':
+                            targetCommitment = connection.commitment;
+                            break;
+                        default:
+                            targetCommitment = 'finalized';
+                    }
+                    await Promise.all([
+                        this.performReauthorization(wallet, authToken),
+                        (async () => {
+                            if (transaction.recentBlockhash == null) {
+                                const preflightCommitment = options?.preflightCommitment;
+                                const preflightCommitmentScore =
+                                    preflightCommitment === 'finalized'
+                                        ? 2
+                                        : preflightCommitment === 'confirmed'
+                                        ? 1
+                                        : 0;
+                                const targetCommitmentScore =
+                                    targetCommitment === 'finalized' ? 2 : targetCommitment === 'confirmed' ? 1 : 0;
+                                const { blockhash } = await connection.getLatestBlockhash({
+                                    commitment:
+                                        preflightCommitment && preflightCommitmentScore < targetCommitmentScore
+                                            ? preflightCommitment
+                                            : targetCommitment,
+                                });
+                                transaction.recentBlockhash = blockhash;
+                            }
+                        })(),
+                    ]);
+                    if (transaction.feePayer == null && this.publicKey) {
+                        transaction.feePayer = this.publicKey;
+                    }
                     const signatures = await wallet.signAndSendTransactions({
-                        connection,
-                        fee_payer: transaction.feePayer,
+                        commitment: targetCommitment,
+                        preflight_commitment: targetCommitment,
```

Comment:
should this be `preflightCommitment`, or always use the same for both?

Reply:
Damnit.

Reply:
Fixed in [200b4bd](https://github.com/solana-mobile/mobile-wallet-adapter/commit/200b4bdccae240a9de8ea5486983654858c2fa5b).